### PR TITLE
Update basic templates default Java version to JDK 17

### PR DESCRIPTION
**Pull request by gdams**
Going forwards Jenkins will only support JDK17+ (https://www.jenkins.io/blog/2024/06/11/require-java-17/)

For Windows I switched from AdoptOpenJDK to Temurin (the new name for the project)

For Ubuntu I've changed to explicitly stating JDK17

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/resources/scripts/windowsInstallJavaScript.ps1 [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
 
-choco install -y adoptopenjdk11
+choco install -y temurin17
```
Comment (timja) [inline src/main/resources/scripts/windowsInstallJavaScript.ps1:3]: Looks like these scripts were broken by chocolatey 2, I'm testing a fix atm

Comment (gdams) [reply src/main/resources/scripts/windowsInstallJavaScript.ps1:3]: I had considered switching this to use winget:

```bash
winget install EclipseAdoptium.Temurin.17.JDK
```

Comment (timja) [reply src/main/resources/scripts/windowsInstallJavaScript.ps1:3]: winget is only available from window server 2022+ as far as I remember

Comment (gdams) [reply src/main/resources/scripts/windowsInstallJavaScript.ps1:3]: The alternative would be to use the Adoptium API directly

```ps
# Set the URL for the Adoptium API
$downloadUrl = "https://api.adoptium.net/v3/installer/latest/17/ga/windows/x64/jdk/hotspot/normal/eclipse"

# Specify the download location
$destination = "$env:TEMP\temurin17.msi"

# Download the JDK installer
Invoke-WebRequest -Uri $downloadUrl -OutFile $destination

# Install the JDK
Start-Process -FilePath msiexec.exe -ArgumentList "/i", $destination, "/quiet", "/norestart" -Wait

# Clean up the installer file
Remove-Item $destination
```

Comment (timja) [reply src/main/resources/scripts/windowsInstallJavaScript.ps1:5]: Probably ok but for now I've pinned chocolatey to 1.x and it worked:

image

File: src/main/java/com/microsoft/azure/vmagent/util/Constants.java [hunk 2]
```diff
@@ -128,7 +128,6 @@ public final class Constants {
     public static final String WINDOWS_SERVER_2022 = "Windows Server 2022";
     public static final String WINDOWS_SERVER_2019 = "Windows Server 2019";
     public static final String WINDOWS_SERVER_2016 = "Windows Server 2016";
-    public static final String UBUNTU_1604_LTS = "Ubuntu 16.04 LTS";
```
Comment (timja) [inline src/main/java/com/microsoft/azure/vmagent/util/Constants.java:131]: `openjdk-17-jdk` is not available on Ubuntu 16, given its so far out of support I've removed it from the list.

Comment (gdams) [reply src/main/java/com/microsoft/azure/vmagent/util/Constants.java:131]: make sense to me

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
Thanks!

**Comment by gdams** [comment]
> Thanks!

when do you expect we will have a new release of this plugin?

**Comment by timja** [comment]
next 5-10 mins or so i'd expect, its automated.

**Pull request merged**